### ISSUE TITLE
[@sucrase/jest-plugin]: Jest: a transform must export something.

### ISSUE THREAD

**Issue by steschwa** [alangpierce/sucrase#638]
There's a demo repository which reproduces the error: https://github.com/steschwa/jest-sucrase-demo

## Setup

**package.json**
```json
{
  "devDependencies": {
    "@sucrase/jest-plugin": "^2.1.0",
    "jest": "^27.0.6"
  }
}
```

**jest.config.js**
```js
module.exports = {
  transform: {
    ".tsx?$": "@sucrase/jest-plugin"
  }
}
```

## Problem

After writing some tests and executing them with `yarn jest`, Jest exits almost immediately with 

> Jest: a transform must export something.

After digging around in the `@jest/transformer` and `jest-util` package i've found out that they require a `export default` for custom transformers: https://github.com/facebook/jest/blob/bc50e7f360ab1845abbaa0b3ad788caead0d3174/packages/jest-util/src/requireOrImportModule.ts#L21.

The problem with `@sucrase/jest-plugin` is, that the published file (`dist/index.js`) has a `{ __esModule: true }` which Jest reads in https://github.com/facebook/jest/blob/bc50e7f360ab1845abbaa0b3ad788caead0d3174/packages/jest-util/src/interopRequireDefault.ts#L10 and makes it think that there is a default export but obviously there's not.

## Possible Solution

A simple solution would be to change `export function process(...)` in https://github.com/alangpierce/sucrase/blob/main/integrations/jest-plugin/src/index.ts to `export default { process: function process(...) }`.

Not sure if thats the way to go but it is a quick & easy fix for the problem.

Would you like me to open a PR?

**Comment by vpontis** [comment]
Yep! So I checked this out in a [test package](https://www.npmjs.com/package/@luma-team/sucrase-jest-plugin-test/access) and I think @steschwa is correct. 

Right now the jest plugin generates:

```js
exports.default = process;
```

But jest wants:

```js
module.exports = { process };

// OR 

exports.default = { process };
```

Btw, from some issues in other repos, I think this is due to a new change in Jest but I can't find the change in Jest's changelog.

**Comment by alangpierce** [comment]
Thanks for the report, and sorry for the delay! I merged the PR from @vpontis and released it as `@sucrase/jest-plugin@2.1.1`.
